client/src/components/Signup/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import {USER_REGISTRATION, registerUser} from './actions'
+
+class Signup extends Component {
+    constructor(props) {
+        super(props)
+        this.handleRegistration = this.handleRegistration.bind(this)
+        this.handleChange = this.handleChange.bind(this)
+        this.state = {
+            name: "",
+            age: "",
+            email: "",
+            password: "",
+            address: "",
+        }
+    }
+
+    handleChange(e) {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+
+    handleRegistration(e) {
+        e.preventDefault()
+        // Here we shall dispatch the action for registration
+        this.props.registerUser(this.state)
+    }
+
+    render() {
+        return (
+            <form onSubmit={this.handleRegistration}>
+                <input
+                    name="name"
+                    type="text"
+                    required={true}
+                    value={this.state.name}
+                    placeholder="First and Last Name"
+                    onChange={this.handleChange}
+                />
+                <input
+                    name="age"
+                    type="number"
+                    value={this.state.age}
+                    placeholder="Your age"
+                    onChange={this.handleChange}
+                />
+                <input
+                    name="email"
+                    type="email"
+                    required={true}
+                    value={this.state.email}
+                    placeholder="Your preferred email address"
+                    onChange={this.handleChange}
+                />
+                <input
+                    name="password"
+                    type="password"
+                    required={true}
+                    value={this.state.password}
+                    placeholder="Password"
+                    onChange={this.handleChange}
+                />
+                <button type="submit"> Register Now </button>
+            </form>
+        )
+    }
+}
+
+function mapStateToProps(state) {
+    console.log('state')
+    console.log(state)
```

Comment:
Is this required? If yes, can we join this into one statement? 
```suggestion
    console.log('state', state)
```

Reply:
This needs to be removed. It was added for debugging purpose. Sorry about this.